### PR TITLE
Node-token-validation handle network errors

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR adds network error handling to `HttpClient`, allowing 2 retries when response returned indicates request timeout or service unavailable.

Note: Additional changes will be made to `HttpClient` in the future to handle proxy requests, which will not be included in this PR.

### CODE REVIEW

File: lib/node-token-validation/src/network/HttpClient.ts [hunk 1]
```diff
@@ -50,12 +45,55 @@ export class HttpClient implements INetworkModule {
...
+
+    /**
+     * Helper function to handle network retries for Axios requests.
+     * Function will retry axios request twice when response is returned with HTTP status code indicating request timeout or service unavailable. 
+     * 
+     * @param {AxiosRequestConfig} request Axios request
+     * @param {string} url URL for request
+     * @returns 
+     */
+    private async sendAxiosRequest<T>(request: AxiosRequestConfig, url: string): Promise<NetworkResponse<T>> {
```
Comment (Robbie-Microsoft) [inline lib/node-token-validation/src/network/HttpClient.ts:58]: This is really neat. Are there any plans to add this functionality to MSAL-Node?

**Review by Robbie-Microsoft — APPROVED** [review]

**Review by bmahall — APPROVED** [review]

File: lib/node-token-validation/src/network/HttpClient.ts [hunk 2]
```diff
@@ -50,7 +45,50 @@ export class HttpClient implements INetworkModule {
...
+        let response;
+        const maxNetworkAttempts = 3;
+        const maxRetryAttempts = 2;
```
Comment (peterzenz) [inline lib/node-token-validation/src/network/HttpClient.ts:61]: Change for a future PR would be to make the max attempts configurable.

File: lib/node-token-validation/src/utils/Constants.ts [hunk 3]
```diff
@@ -38,3 +38,11 @@ export const JwtConstants = {
 export const TokenType = {
     JWT: "JWT"
 };
+
+/**
+ * HTTP status code constants
```
Comment (peterzenz) [inline lib/node-token-validation/src/utils/Constants.ts:43]: More documentation needed here on what you're using them for.  Something to the effect of which failures we'll optionally attempt a retry on and for what flows specifically we'll attempt this retry.

**Review by peterzenz — APPROVED** [review]
Minor comment changes and something for a future PR.

**Pull request merged**